src/AutoRest.CSharp/Output/Models/RestClient.cs

```diff
@@ -459,7 +459,18 @@ PathSegment TextSegment(string text)
 
             foreach ((string text, bool isLiteral) in StringExtensions.GetPathParts(httpRequestUri))
             {
-                yield return isLiteral ? TextSegment(text) : parameters[text];
+                if (isLiteral)
+                {
+                    yield return TextSegment(text);
+                }
+                else
+                {
+                    if (!parameters.ContainsKey (text))
+                    {
+                        throw new KeyNotFoundException ($"\n\nError while processing request {httpRequestUri}: key '{text}' was undefined.\nVerify that it is correctly defined, and if not please file an issue at https://github.com/Azure/autorest.csharp/issues/new with a swagger that reproduces.\n");
```

Comment:
Let's be a bit more specific here. Something like "parameter is used in the path template but is not defined in the parameters collection"